### PR TITLE
add doc for make package

### PR DESCRIPTION
**Pull request by cpanato**
jira: https://mattermost.atlassian.net/browse/MM-11446

### CODE REVIEW

**Comment by jasonblais** [comment]
This is pretty much identical to my PR (https://github.com/mattermost/mattermost-developer-documentation/pull/141) but @crspeller mentioned `There is a bunch wrong with what you have written there. If we want to support using the package target, let's assign this to a dev.`

So will defer to @crspeller  to review

**Comment by cpanato** [comment]
It is the same :-)

**Comment by jasonblais** [comment]
Ah - @cpanato did you try if it worked for you?

**Comment by cpanato** [comment]
will simplify the text

**Comment by cpanato** [comment]
I did the test here and it is easier, I just ran `make package` after I build the webapp and got all packages (linux, win, osx)

File: site/content/contribute/server/developer-workflow.md [hunk 1]
```diff
@@ -41,14 +41,18 @@ Some useful `make` commands:
 * `make run-fullmap` will run the server and build the client with the full source map for easier debugging
 * `make clean` cleans your local environment of temporary files
 * `make nuke` wipes your local environment back to a completely fresh start
+* `make package` creates packages for distributing your builds and puts them in the `~/go/src/github.com/mattermost/mattermost-server/dist` directory. You need to build the `mattermost-webapp` first and the you can package the app.
```
Comment (crspeller) [inline site/content/contribute/server/developer-workflow.md]: You need to run `make build` and `make build-webapp` before this.

Comment (cpanato) [reply site/content/contribute/server/developer-workflow.md]: done

**Pull request merged**